Automattic/woocommerce-payments
# Toggle terms for save payments in UPE block checkout

Fixes #2605

#### Changes proposed in this Pull Request

This change adds the legal mandate text to the UPE block checkout when the "Save payment information to my account for future purchases." checkbox is selected. The mandate should always be shown when a subscription product is in the cart. Otherwise, it should be shown when the save payment method checkbox is toggled.

<img width="827" alt="Screen Shot 2021-11-11 at 4 08 55 PM" src="https://user-images.githubusercontent.com/6216000/141284690-8b9494a2-5d39-41c4-9c62-4240410c7cb3.png">









#### Testing instructions

1. Enable the UPE at checkout using the WCPay Dev Tools.
2. Add the WooCommerce Blocks plugin
3. As a customer, add a subscription product to the cart and proceed to a blocks checkout. Note that the above legal mandate is present underneath the payment fields inside the UPE.
4. Remove any subscription products from the cart and add a regular product. Proceed to the blocks checkout and note that when the save payment checkbox is checked, the legal mandate appears. Confirm that this mandate disappears if the checkbox is unchecked







-------------------


**Post merge**
